### PR TITLE
Refactor integration and system tests to remove Zookeeper, add group protocol parameter

### PR DESCRIPTION
**Pull request by kirktrue**
Removing "zk" from the list of quorum values to test in the integration and system tests. This does not remove all occurrences as not all tests support KRaft.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala [hunk 1]
```diff
@@ -66,7 +66,7 @@ class ListOffsetsIntegrationTest extends KafkaServerTestHarness {
   }
 
   @ParameterizedTest
-  @ValueSource(strings = Array("zk", "kraft"))
+  @ValueSource(strings = Array("kraft"))
   def testListMaxTimestampWithEmptyLog(quorum: String): Unit = {
```
Comment (AndrewJSchofield) [inline core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala:70]: Isn't the proper change here to remove the quorum parameter entirely?

**Review by AndrewJSchofield — CHANGES_REQUESTED** [review]

File: core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala (hunk 1 above)
Comment (kirktrue) [reply core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala:70]: I'd wondered that and wanted someone to raise the question.

I think it makes sense to remove it.

Comment (AndrewJSchofield) [reply core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala:70]: Happy to oblige :)

Comment (kirktrue) [reply core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala:70]: I have a question out to @cmccabe regarding his preference.

Comment (kirktrue) [reply core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala:70]: `TestInfoUtils` uses the parameters to determine if it `isKraft()`, `isZkMigrationTest()`, etc. Usages of those methods would need to be updated too.

Comment (AndrewJSchofield) [reply core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala:70]: Yes, I'm sure, but wouldn't AK 4.0 have all of the ZK migration tests removed, and wouldn't all tests be ZK?

**Comment by kirktrue** [comment]
@chia7712—do you have any advice for the best way to arrange these changes for review?

**Comment by kirktrue** [comment]
@mumrah—can you take a look at approach in this PR and let me know if it's correct? Thanks!

File: core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala (hunk 1 above)
Comment (mumrah) [reply core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala:70]: There are some shenanigans done in the base test classes regarding the `quorum` param. Until we remove the ZK bits from that test infra, I think we actually need the `quorum` param :-/

**Review by mumrah — COMMENTED** [review]
@kirktrue the approach in the ducktape tests seems fine to me. 

You might consider doing the system test changes in a separate PR since this one has grown quite large.

File: core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala (hunk 1 above)
Comment (kirktrue) [reply core/src/test/scala/integration/kafka/admin/ListOffsetsIntegrationTest.scala:70]: I approve of the use of "shenanigans" to describe this 👍

**Comment by kirktrue** [comment]
@mumrah @cmccabe—is there an existing Jira that PR overlaps with the changes in this PR? I want to make sure I'm not duplicating efforts and/or stepping on anyone’s work.

Ideally there would be two PRs: one for the integration tests and one for the system tests, as @mumrah suggested above.

Thanks!

**Comment by kevin-wu24** [comment]
> @mumrah @cmccabe—is there an existing Jira that PR overlaps with the changes in this PR? I want to make sure I'm not duplicating efforts and/or stepping on anyone’s work.
> 
> Ideally there would be two PRs: one for the integration tests and one for the system tests, as @mumrah suggested above.
> 
> Thanks!

Your system test changes are a superset of the changes I made in https://github.com/apache/kafka/pull/17638/files (involves just removing explicit `quorum.zk` parameterizations), which is tracked by https://issues.apache.org/jira/browse/KAFKA-17625. When I tried making the `metadata_quorum` default value to use KRaft I ran into some issues getting the tests to work properly, so I created some separate JIRA issues for those tests (The Kafka Streams work is already being tracked here: https://issues.apache.org/jira/browse/KAFKA-17609): 

- https://issues.apache.org/jira/browse/KAFKA-17915
- https://issues.apache.org/jira/browse/KAFKA-17916
- https://issues.apache.org/jira/browse/KAFKA-17917

My PR also removes `zk` from `quorum.all_non_upgrade` and `quorum.all` so those ZK test variations would be removed, but I don't see that in this PR's diff.

**Comment by kirktrue** [comment]
Closing this PR in favor of #17669 and #17670.

**Pull request closed without merging**